docs/modules/extensions/pages/conversion-process-overview.adoc

```diff
@@ -11,12 +11,13 @@ For example, a sub-section of the AST representing a title with a paragraph will
 NOTE: Some liberty is taken to make the process easier to understand.
 In reality, Asciidoctor has implementation details that divert from the 3 steps above.
 
-The different extension types are called in different steps of the conversion process in the order shown in the table.
-Within each type:
+The different extension types are called in different steps of the conversion process in the following order:
 
-* Prepocessors is called just before parsing.
-* Treeprocessors is called right before processing.
-* Block macro, inline macro, and block processors are called during processing in the order that they appear in the document.
-* Postprocessors are called before output generation.
-* Include and DocinfoProcessors are called in an arbitrary and changeable order during processing.
-In the case of Include processors, it is called whenever an `include::` is found.
+. IncludeProcessors are called in an arbitrary and changeable order during processing and are called whenever an `include::` is found.
+. Preprocessors are called just before parsing.
+. BlockMacroProcessors are called during processing in the order that they appear in the document.
+. BlockProcessors are called during processing in the order that they appear in the document.
+. Treeprocessors are called right before processing.
+. InlineMacroProcessors are called during processing in the order that they appear in the document.
+. Postprocessors  are called before output generation.
```

Comment:
Silly double space typo, sorry :nerd_face:

Reply:
Hah :D If it's only that

Reply:
Hope you dont mind me jumping in? The critical information for me is that BlockProcessors and BlockMacroProcessors are interleaved in document order, then InlineMacroProcessors are run. Your revision says nothing about the relative ordering of the three types.

Reply:
Agreed. It should be emphasized that inline macro processors are called in a different phase, the convert phase, though still in document order.

Reply:
> Agreed. It should be emphasized that inline macro processors are called in a different phase, the convert phase, though still in document order.

That's right, back when I wrote that I did not have the insight I have now. I tried to make an analogy with standard compilers with parsing, processing & convert phases and that's not accurate. I assume the correct would be to talk about preprocessing, parsing and converting?

For context, and to see if we should redirect the explanation...when I wrote this I wanted to answer some user questions I've seen and I myself have:
* Do I have acces to the AST and can I manipulate it the this extension?
* Can I output formatted content (e.g HTML or text) from this extension?
* Why if inline macro processor is run during convert can I inject AST nodes? Like in example https://docs.asciidoctor.org/asciidoctorj/latest/extensions/inline-macro-processor/

But going for a "yes/no" seemed too complex, so I mapped the extensions to the phases explaining what info is available in each phase.

Reply:
It's definitely a bit tricky. The reason an inline macro can add AST nodes is because parsing for inline content is happening during conversion. So it's really a "just in time" AST mutation. Going even deeper, an inline macro doesn't actually add a node to the AST. Rather, the return value is immediately converted and added into the conversion stream.

Keep in mind, this is likely going to change in the spec / Austen...which will move inline processing into the parsing phase (eventually).